src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1411 @@
+    grid-template-rows: min-content 1fr min-content;
+    max-height: 360px;
+  }
+
+  .body {
+    cursor: auto;
+    display: grid;
+    gap: 8px;
+    overflow: auto;
+    text-align: left;
+  }
+
+  .values {
+    display: flex;
+    flex-direction: column;
+  }
+
+  .value {
+    border-radius: 0;
+    justify-content: left;
+    padding: 16px 8px;
+  }
+
+  .value:hover {
+    background: var(--st-gray-20);
+  }
+
+  .value.active,
+  .value.active:hover {
+    background: #4fa1ff4f;
+  }
+
+  .body :global(.input-inline) {
+    padding: 0;
+  }
+
+  .list-buttons {
+    display: flex;
+    flex-direction: column;
+    gap: 8px;
+    padding: 8px;
+    width: 100%;
+  }
+
+  .empty-state {
+    margin: 8px;
+  }
+
+  .menu-border-top {
+    border-top: 1px solid var(--st-gray-20);
+  }
+  .external-event-form-container {
+    display: grid;
+    grid-template-rows: min-content auto;
+    height: 100%;
+    overflow: hidden;
+  }
+
+  .external-event-form {
+    overflow-y: auto;
+  }
+
+  .external-event-directive-definition {
+    padding: 0.5rem;
+  }
+
+  .external-event-header {
+    align-items: center;
+    background: var(--st-gray-10);
+    border-bottom: 1px solid var(--st-gray-15);
+    display: flex;
+    flex-shrink: 0;
+    font-style: italic;
+    padding: 4px 8px;
+    padding-left: 8px;
+  }
+
+  .external-event-header-icons {
+    align-items: center;
+    display: flex;
+  }
+
+  .external-event-error-rollup {
+    display: inline;
+    font-style: normal;
+  }
+
+  .external-event-header-title-placeholder,
+  .external-event-header-title-value {
+    word-break: break-word;
+  }
+
+  .external-event-header-title-value {
+    overflow: hidden;
+    padding: 4px 0px;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    word-break: break-all;
+  }
+
+  .external-event-header-title-placeholder {
+    padding: 4px 8px;
+  }
+
+  .external-event-header-title {
+    align-items: flex-start;
+    border-radius: 4px;
+    display: flex;
+    width: 100%;
+  }
+
+  .external-event-header-title :global(fieldset) {
+    padding: 0;
+    width: 100%;
+  }
+
+  .external-event-header-title-edit-button:hover {
+    background-color: var(--st-white);
+  }
+
+  .external-event-header-title--editing {
+    gap: 8px;
+    padding: 0;
+    width: 100%;
+  }
+
+  .external-event-header-changelog {
+    border: 1px solid transparent;
+    display: flex;
+    width: 24px;
+  }
+
+  .external-event-header-changelog:hover {
+    color: #007bff;
+  }
+
+  .revision-preview-header {
+    align-items: center;
+    background-color: #e6e6ff;
+    border-bottom: 1px solid #c4c6ff;
+    border-top: 1px solid #c4c6ff;
+    display: flex;
+    flex-shrink: 0;
+    justify-content: space-between;
+    padding: 4px 8px;
+    padding-left: 8px;
+  }
+
+  .annotations {
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+  }
+
+  .btn-group {
+    align-self: center;
+    margin-left: auto;
+  }
+
+  #file-upload-field {
```

Comment:
Generally we try to avoid using IDs in CSS in aerie unless strictly necessary. Classes are preferred.